src/containers/ServiceAccount/ServiceAccount.js

```diff
@@ -209,7 +209,13 @@ export /* istanbul ignore next */ class ServiceAccountContainer extends Componen
     let rowsForImgPull = [];
     if (serviceAccount.imagePullSecrets && !loading) {
       rowsForImgPull = serviceAccount.imagePullSecrets.map(({ name }) => {
-        const { type } = secrets.find(secret => secret.name === name);
+        const foundSecret = secrets.find(secret => secret.name === name);
+        let type = '';
+        if (!foundSecret) {
+          type = 'unknown-type';
+        } else {
+          type = { type };
```

Comment:
should this be `({ type } = foundSecret);` to set it to the value of the type field from the found secret instead of an object containing the type?

see previous comment for an alternative that avoids this awkward syntax for destructuring to a previously declared variable

Reply:
Done, that's so much neater - figuring out the root cause now (and helpfully this branch unlocks Megan to fix the other issue I spotted)